compiler/rustc_hir_typeck/src/method/suggest.rs

```diff
@@ -3107,6 +3109,7 @@ fn print_disambiguation_help<'tcx>(
     err: &mut Diagnostic,
     trait_name: String,
     rcvr_ty: Ty<'_>,
+    self_ty: Ty<'_>,
```

Comment:
This is the self type of the method's impl? It should have a better name, imo.

Reply:
What would be a better name? `impl_self_ty` or something?

Reply:
Yeah that works